cve_bin_tool/parsers/go.py

```diff
@@ -27,9 +78,23 @@ def run_checker(self, filename):
                     packages = False
                     continue
                 if packages:
-                    product = line.split(" ")[0].split("/")[-1]
-                    version = line.split(" ")[1][1:].split("-")[0].split("+")[0]
-                    vendor = self.find_vendor(product, version)
-                    if vendor is not None:
-                        yield from vendor
+                    parts = line.split(" ")
+                    if len(parts) >= 2:
+                        product = line.split(" ")[0].split("/")[-1]
+                        version = line.split(" ")[1][1:].split("-")[0].split("+")[0]
+                        vendors = self.find_vendor(product, version)
+                        if vendors is not None:
+                            for v in vendors:
+                                if self.is_valid_data(
+                                    product, version, v.product_info.vendor
+                                ):
+                                    self.generate_purl(
+                                        product,
```

Comment:
I think this'll be a bit more clear if we remove a few parameters, but a rule of thumb: if it's got more than 3 parameters, it's worth putting the names in to make it clearer what they are.  e.g. below you'd have `subpath=None` if we weren't going to set that to a default value in the function so it could be skipped entirely.